katsdptelstate/aio/test/test_telescope_state.py

```diff
@@ -31,520 +33,512 @@
         """Calling wait_key with a condition that is met must return (indexed version)."""
-        await self.ts.set_indexed('test_key', 'idx', 5)
-        await self.ts.wait_key('test_key', lambda v, t: v == {'idx': 5} and t is None)
+        await ts.set_indexed('test_key', 'idx', 5)
+        await ts.wait_key('test_key', lambda v, t: v == {'idx': 5} and t is None)
 
-    async def test_wait_key_timeout(self) -> None:
+    async def test_wait_key_timeout(self, ts: TelescopeState) -> None:
         """wait_key must time out in the given time if the condition is not met"""
-        with self.assertRaises(asyncio.TimeoutError):
-            with async_timeout.timeout(0.1):
-                await self.ts.wait_key('test_key')
-        with self.assertRaises(asyncio.TimeoutError):
-            with async_timeout.timeout(0.1):
+        with pytest.raises(asyncio.TimeoutError):
+            async with async_timeout.timeout(0.1):
+                await ts.wait_key('test_key')
+        with pytest.raises(asyncio.TimeoutError):
+            async with async_timeout.timeout(0.1):
                 # Takes a different code path, even though equivalent
-                await self.ts.wait_key('test_key', lambda value, ts: True)
+                await ts.wait_key('test_key', lambda value, ts: True)
 
-    async def _set_key_immutable(self) -> None:
+    async def _set_key_immutable(self, ts: TelescopeState) -> None:
         await asyncio.sleep(0.1)
-        await self.ts.set('test_key', 123)
+        await ts.set('test_key', 123)
 
-    async def _set_key_mutable(self) -> None:
+    async def _set_key_mutable(self, ts: TelescopeState) -> None:
         await asyncio.sleep(0.1)
-        await self.ts.add('test_key', 123, ts=1234567890)
+        await ts.add('test_key', 123, ts=1234567890)
 
-    async def _set_key_indexed(self) -> None:
+    async def _set_key_indexed(self, ts: TelescopeState) -> None:
         await asyncio.sleep(0.1)
-        await self.ts.set_indexed('test_key', 'idx', 123)
+        await ts.set_indexed('test_key', 'idx', 123)
 
-    async def test_wait_key_delayed(self) -> None:
+    async def test_wait_key_delayed(self, ts: TelescopeState) -> None:
         """wait_key must succeed with a timeout that does not expire before the condition is met."""
         for (set_key, value, timestamp) in [
                 (self._set_key_mutable, 123, 1234567890),
                 (self._set_key_immutable, 123, None),
                 (self._set_key_indexed, {'idx': 123}, None)]:
-            task = asyncio.ensure_future(set_key())
-            await self.ts.wait_key('test_key', lambda v, t: v == value and t == timestamp)
-            self.assertEqual(value, await self.ts.get('test_key'))
+            task = asyncio.ensure_future(set_key(ts))
+            await ts.wait_key('test_key', lambda v, t: v == value and t == timestamp)
+            assert value == await ts.get('test_key')
             await task
```

Comment:
So here we wait asynchronously for `test_key` to appear, and only then add the key?

Reply:
Newer versions of async_timeout give a DeprecationWarning without it (and pytest shows DeprecationWarnings). I must check if the requirements depend on a new enough version for the `async with` version.

Reply:
That's what it looks like to me.